File: src/main/java/com/example/interviewlandbackend/model/Question.java
package com.example.interviewlandbackend.model;

import javax.persistence.*;
import java.io.Serializable;


@Entity
@Table(name = "questions")
public class Question implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String questionTitle;

    @Column(columnDefinition="TEXT")
    @Lob
    private String description;

    @Column(columnDefinition="TEXT")
    @Lob
    private String sourceCode;


    @ManyToOne
    @JoinColumn(name = "section_id")
    private Section section;


    public Question() {
    }

    public Question(String questionTitle, String description, String sourceCode, Section section) {
        this.questionTitle = questionTitle;
        this.description = description;
        this.sourceCode = sourceCode;
        this.section = section;
    }


    public int getId() {
        return id;
    }

    public String getQuestionTitle() {
        return questionTitle;
    }

    public String getDescription() {
        return description;
    }

    public String getSourceCode() {
        return sourceCode;
    }

    public Section getSection() {
        return section;
    }


    public void setQuestionTitle(String questionTitle) {
        this.questionTitle = questionTitle;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setSourceCode(String sourceCode) {
        this.sourceCode = sourceCode;
    }

    public void setSection(Section section) {
        this.section = section;
    }
}

File: src/main/java/com/example/interviewlandbackend/model/Question.kt
package com.example.interviewlandbackend.model

import java.io.Serializable
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "questions")
data class Question @JvmOverloads constructor(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int?=0,

    var questionTitle: String,

    @Column(columnDefinition = "TEXT")
    @Lob
    var description: String,

    @Column(columnDefinition = "TEXT")
    @Lob
    var sourceCode: String,

    @ManyToOne
    @JoinColumn(name = "section_id")
    var section: Section,

    val date: Date?=Date(System.currentTimeMillis())


):Serializable
